File: #2016.maximum-difference-between-increasing-elements.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        
        int front = nums.front();
        int result = 0;
        for (const int &it : nums) {
            front = std::min(front, it);
            result = std::max(result, it - front);
        }
        return result == 0 ? -1 : result;
    }
};

File: #2016.maximum-difference-between-increasing-elements.ts
function maximumDifference(nums: number[]): number {
    let min = Infinity;
    return nums.reduce((acc, t) => {
        if (t - min > acc && t - min !== 0) {
            acc = t - min;
        }

        if (min > t) {
            min = t;
        }
        
        return acc;
    }, -1);
};
